### DISCUSSION TITLE
Link IRL time to a token for chat bot memory.

### DISCUSSION THREAD

**Discussion by xzuyn** [LostRuins/koboldcpp#86] [Ideas]
So you could have something in the memory for a chatbot like [Character: ChatBot; Species: AI; Age: 4 days; Birthday: April 13, 2023; Current Date: April 17, 2023], and have the current date get changed every time you send a prompt. The age could also be calculated if a birthday is set.

Although it may act funny if you ask it about the current date because now the memory will have a date that's different from what it said later. Maybe it could replace any mentions of the date with the one from the token?

Not sure how this would be added.

**Comment by LostRuins** [comment]
So ChatGPT actually does that, and you can do it too. KoboldCpp provides a friendly, easy to use API that you can build on top off, compatible with a subset of the official Kobold API (reference here: https://link.concedo.workers.dev/koboldapi)

Simply write a script that fetches a few things such as the current date and time, locations, weather etc, and add these information before the context you submit to the API.
